src/transformers/configuration_utils.py

```diff
@@ -132,6 +132,10 @@ def save_pretrained(self, save_directory):
             save_directory (:obj:`string`):
                 Directory where the configuration JSON file will be saved.
         """
+        if os.path.isfile(save_directory):
+            raise AssertionError("Provided path ({}) should be a directory, not a file".format(save_directory))
+        if not os.path.exists(save_directory):
+            os.path.makedirs(save_directory)
```

Comment:
You should remove all those `if not...: os.makedirs()` and replace by `os.path.makedirs(exist_ok=True)` otherwise you can have race-condition crashes when calling from multiple processes concurrently

Reply:
Should have read this comment first :D 
Yeah, I agree to use `exist_ok=True` everywhere